### PR TITLE
Upgrading to Laravel 5.4

### PR DESCRIPTION
**Pull request by weerd**
### What does this PR do?
I'd like to try a piecemeal process of performing all the Laravel upgrade steps to move from 5.3 to 5.4 (and then eventually from 5.4 to 5.5). 

I've added an upgrade checklist to indicate what steps in the upgrade process have been completed based on the [upgrade guide](https://laravel.com/docs/5.4/upgrade) in the Laravel documentation. With this approach others can take over when they have some free time and complete a step in the process and eventually we'll do the full upgrade.

This is a little experiment to see if we can upgrade Laravel as a team!
go team

There's lots of useful stuff in Laravel 5.4 and also in 5.5 that would be great to have access too. Specifically it'd be great to be able to use the file upload faker for testing!

When reviewing the code changes, it might be best to go [commit-by-commit](https://github.com/DoSomething/phoenix-next/pull/607/commits) which easily follows the checklist items below!

### What are the relevant tickets/cards?
Refs https://www.pivotaltracker.com/story/show/154419780

### Checklist:
- [x] Updating Dependencies
- [x] Authorization (not applicable)
- [x] Blade (not applicable)
- [x] Bootstrappers (not applicable)
- [x] Broadcasting
- [x] Collections (not applicable)
- [x] Container (not applicable)
- [x] Console (not applicable)
- [x] Database (not applicable)
- [x] Eloquent (not applicable)
- [x] Events (not applicable)
- [x] Exceptions (not applicable)
- [x] Mail (not applicable)
- [x] Queue (not applicable)
- [x] Redis (not applicable, not using clusters)
- [x] Routing
- [x] Sessions (not applicable)
- [x] Testing
- [x] Translation
- [x] URL Generation
- [x] Validation
- [x] Miscellaneous

### CODE REVIEW

**Comment by weerd** [comment]
I recommend committing after each main "step", checking off the item and using a commit message like...
`"Updating Blade step completed."`

**Comment by DFurnes** [comment]
Rebasing to fix the `composer.lock` conflict!

**Comment by DFurnes** [comment]
Knocked out a few more steps!

**Comment by weerd** [comment]
I'll address the wercker test errors after a few other items I need to get done today :)

File: public/index.php [hunk 1]
```diff
@@ -15,7 +15,7 @@
 | Composer provides a convenient, automatically generated class loader for
 | our application. We just need to utilize it! We'll simply require it
 | into the script here so that we don't have to worry about manual
-| loading any of our classes later on. It feels nice to relax.
+| loading any of our classes later on. It feels great to relax.
```
Comment (mendelB) [inline public/index.php]: 🍹

Comment (weerd) [reply public/index.php]: haha, yeah it's in the official diff!

**Comment by weerd** [comment]
Imports on tests all fixed and looks like all checks passed! @mendelB @DFurnes give it a quick looksie and if everything seems 👌 , lets merge and get this on Phoenix-Preview 🎉

**Comment by mendelB** [comment]
Lookin' 💯

File: tests/BrowserKitTestCase.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+use App\Exceptions\Handler;
+use App\Repositories\CampaignRepository;
+use Illuminate\Contracts\Console\Kernel;
+use DoSomething\Gateway\Testing\WithOAuthTokens;
+use Illuminate\Contracts\Debug\ExceptionHandler;
+
+abstract class BrowserKitTestCase extends \Laravel\BrowserKitTesting\TestCase
+{
+    use DatabaseSetup, WithOAuthTokens;
```
Comment (DFurnes) [inline tests/BrowserKitTestCase.php]: Should this test `use CreatesApplication` too?

Comment (weerd) [reply tests/BrowserKitTestCase.php]: yep! good catch 💃

**Pull request merged**